integration/spark/src/main/java/org/apache/spark/sql/secondaryindex/query/SecondaryIndexQueryResultProcessor.java

```diff
@@ -277,15 +277,10 @@ private void processResult(List<CarbonIterator<RowBatch>> detailQueryResultItera
       CarbonDimension dims = dimensions.get(i);
       boolean isComplexColumn = false;
       // As complex column of MainTable is stored as its primitive type in SI,
-      // we need to check if dimension is complex dimension or not based on dimension
-      // name. Check if name exists in complexDimensionInfoMap of main table result
-      if (!complexDimensionInfoMap.isEmpty() && complexColumnParentBlockIndexes.length > 0) {
-        for (GenericQueryType queryType : complexDimensionInfoMap.values()) {
-          if (queryType.getName().equalsIgnoreCase(dims.getColName())) {
-            isComplexColumn = true;
-            break;
-          }
-        }
+      // we need to check if dimension is complex dimension or not based on isParentColumnComplex
+      // property.
+      if (Boolean.parseBoolean(dims.getColumnProperties().get("isParentColumnComplex"))) {
```

Comment:
it may throw NPE. Please check and handle

Reply:
Done